### ISSUE TITLE
In-place update of data

### ISSUE THREAD

**Issue by martindurant** [jakevdp/mpld3#318]
One deficiency of mpld3 in an interactive setting (e.g., embedded in a notebook or web-page), is that to get an update of the figure, one has to redraw the figure. 
Since the machinery is in place to generate paths and glyphs for any figure element, if a function were exposed to get the HTML representation of just a given plot element or collection of elements, a figure could be updated in-place by via, for example, an ajax call to a python server.
This would require being able to call the JS renderer with only those data that have been updated.

**Comment by jakevdp** [comment]
Yes, this would be very nice! Probably the cleanest way to do this would be to use the traitlets system used by IPython, which can mimic JS structures on the Python side. The benefit is that you would basically get IPython widget compatibility for free.

That said, I don't have the bandwidth to tackle such a project at the moment. If you'd like to look into that, contributions are welcome!

**Comment by martindurant** [comment]
I can confirm that it _is_ possible, see
https://dl.dropboxusercontent.com/u/66310054/streaming_mpld3.ipynb
where a python function is called at regular intervals, which updates a line in-place. I have not yet figured out how to halt the updates... (perhaps store the intervalID in the line object?).
Note that this is for Jupyter 4, and presumably won't work for earlier ipython, as things got renamed.
